src/shogun/base/SGObject.h

```diff
@@ -996,6 +988,22 @@ class CSGObject
 	void register_observable(
 		const std::string& name, const std::string& description);
 
+	/**
+	 * Get the current step for the observed values.
+	 */
+#ifndef SWIG
+	SG_FORCED_INLINE int64_t get_step() const
```

Comment:
@karlnapf If I do not guard it against SWIG this will generate an error when building the interfaces. I have no idea why since the method is protected. It may be caused by the fact that it is used inside `put()`.

Reply:
Even if it is protected?

Reply:
Also what’s the error? :)

Reply:
```
/build/opt/include/shogun/base/SGObject.h:994: Error: Syntax error - possibly a missing semicolon.
src/interfaces/python/CMakeFiles/_interface_python.dir/build.make:61: recipe for target 'src/interfaces/python/shogunPYTHON_wrap.cxx' failed
make[2]: *** [src/interfaces/python/shogunPYTHON_wrap.cxx] Error 1
CMakeFiles/Makefile2:86: recipe for target 'src/interfaces/python/CMakeFiles/_interface_python.dir/all' failed
make[1]: *** [src/interfaces/python/CMakeFiles/_interface_python.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

Reply:
it is not protected. SWIG should ignore protected methods. If it is not possible to do that then guarding is fine